extensions/vault/runtime/src/main/java/io/quarkus/vault/config/VaultServerRuntimeConfig.java

```diff
@@ -0,0 +1,172 @@
+    public static final String DEFAULT_CACHE_PERIOD = "10M";
+    public static final String K8S_CACERT = "/var/run/secrets/kubernetes.io/service-account/ca.crt";
+    public static final String DEFAULT_CONNECT_TIMEOUT = "5S";
+    public static final String DEFAULT_READ_TIMEOUT = "1S";
+    public static final String DEFAULT_TLS_SKIP_VERIFY = "false";
+
+    /**
+     * vault url. eg https://localhost:8200
+     */
+    @ConfigItem
+    public Optional<String> url;
+
+    /**
+     * k8s auth role
+     */
+    @ConfigItem
+    public Optional<String> role;
+
+    /**
+     * authentication type when logging in to get a client token. possible values:
+     * - k8s: kubernetes authentication as defined in https://www.vaultproject.io/api/auth/kubernetes/index.html
+     * - userpass (through properties userpassUsername and userpassPassword)
+     */
+    @ConfigItem(defaultValue = "K8S")
+    public VaultAuthType authType;
+
+    /**
+     * location of jwt file containing the k8s token to authenticate against
+     */
+    @ConfigItem(defaultValue = DEFAULT_K8S_TOKEN_PATH)
+    public String k8sJwtPath;
+
+    /**
+     * user for userpass auth method
+     */
+    @ConfigItem
+    public String userpassUsername;
+
+    /**
+     * password for userpass auth method
+     */
+    @ConfigItem
+    public String userpassPassword;
+
+    /**
+     * vault token, bypassing k8s auth
+     */
+    @ConfigItem
+    public Optional<String> token;
+
+    /**
+     * Renew grace period duration. default=1h.
+     * This value if used to extend a lease before it expires its ttl, or recreate a new lease before the current
+     * lease reaches its max_ttl.
+     * By default vault lease_duration is equal to 7 days (ie: 168h or 604800s).
+     * If a connection pool maxLifetime is set, it is reasonable to set the renewGracePeriod to be greater
+     * than the maxLifetime, so that we are sure we get a chance to renew leases before we reach the ttl.
+     * In any case you need to make sure there will be attempts to fetch secrets within the renewGracePeriod,
+     * because that is when the renewals will happen. This particularly important for db dynamic secrets
+     * because if the lease reaches its ttl or max_ttl, the password of the db user will become invalid and
+     * it will be not longer possible to log in.
+     * This value should also be smaller than the ttl, otherwise that would mean that we would try to recreate
+     * leases all the time.
+     */
+    @ConfigItem(defaultValue = DEFAULT_RENEW_GRACE_PERIOD)
+    public Duration renewGracePeriod;
+
+    /**
+     * vault config source cache period
+     */
+    @ConfigItem(defaultValue = DEFAULT_CACHE_PERIOD)
+    public Duration cachePeriod;
+
+    /**
+     * used to hide confidential infos, for logging in particular.
+     * possible values: LOW=display everything, MEDIUM=hide passwords and tokens HIGH=hide lease ids and dynamic credentials
+     * username
```

Comment:
Maybe a list (either in Javadoc or Asciidoc) would be better?

Reply:
done